client/src/connection/itc/index.ts

```diff
@@ -0,0 +1,414 @@
+      this._shellProcess.stdin.write(`$username = "${username}"\n`);
+      const password = await this.fetchPassword();
+      this._shellProcess.stdin.write(`$password = "${password}"\n`);
+      this._shellProcess.stdin.write(
+        `$serverName = "${
+          protocol === ITCProtocol.COM ? "ITC Local" : "ITC IOM Bridge"
+        }"\n`,
+      );
+      this._shellProcess.stdin.write(
+        `$runner.Setup($profileHost,$username,$password,$port,$protocol,$serverName)\n`,
+        this.onWriteComplete,
+      );
+      this._shellProcess.stdin.write(
+        "$runner.ResolveSystemVars()\n",
+        this.onWriteComplete,
+      );
+
+      if (this._config.sasOptions?.length > 0) {
+        const sasOptsInput = `$sasOpts=${this.formatSASOptions(
+          this._config.sasOptions,
+        )}\n`;
+        this._shellProcess.stdin.write(sasOptsInput, this.onWriteComplete);
+        this._shellProcess.stdin.write(
+          `$runner.SetOptions($sasOpts)\n`,
+          this.onWriteComplete,
+        );
+      }
+    }
+
+    // free objects in the scripting env
+    process.on("exit", async () => {
+      close();
+    });
+
+    return setupPromise;
+  };
+
+  private storePassword = async () =>
+    await this._secretStorage.store(this._passwordKey, this._password);
+
+  private clearPassword = async () => {
+    await this._secretStorage.store(this._passwordKey, "");
+    this._password = "";
+  };
+
+  private fetchPassword = async (): Promise<string> => {
+    if (this._config.protocol === ITCProtocol.COM) {
+      return "";
+    }
+
+    const storedPassword = await this._secretStorage.get(this._passwordKey);
+    if (storedPassword) {
+      this._password = storedPassword;
+      return storedPassword;
+    }
+
+    this._password =
+      (await window.showInputBox({
+        ignoreFocusOut: true,
+        password: true,
+        prompt: l10n.t("Enter your password for this connection."),
+        title: l10n.t("Enter your password"),
+      })) || "";
+
+    return this._password;
+  };
+
+  /**
+   * Executes the given input code.
+   * @param code A string of SAS code to execute.
+   * @param onLog A callback handler responsible for marshalling log lines back to the higher level extension API.
+   * @returns A promise that eventually resolves to contain the given {@link RunResult} for the input code execution.
+   */
+  public run = async (code: string): Promise<RunResult> => {
+    const runPromise = new Promise<RunResult>((resolve, reject) => {
+      this._runResolve = resolve;
+      this._runReject = reject;
+    });
+
+    //write ODS output to work so that the session cleans up after itself
+    const codeWithODSPath = code.replace(
+      "ods html5;",
+      `ods html5 path="${this._workDirectory}";`,
+    );
+
+    //write an end mnemonic so that the handler knows when execution has finished
+    const codeWithEnd = `${codeWithODSPath}\n%put ${LineCodes.RunEndCode};`;
```

Comment:
Just saw [this](https://github.com/sassoftware/vscode-sas-extension/discussions/605) today. Perhaps we can move this end code also to powersell?

Reply:
Eventually, I think that's a great idea. This is only _now_ complicated by the fact that we are using async submissions  so that cancel can be enabled. 

My preference is to move this to its own pull request after this code has been merged.